optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/locator/LocationStrategyResolver.java

```diff
@@ -104,8 +120,8 @@ public LocationStrategy determineLocationStrategy(Object object) {
             return null;
         }
         if (memberList.size() > 1) {
-            throw new IllegalStateException("The class (" + clazz
-                    + ") has " +  memberList.size() + " members (" + memberList + ") with a "
+            throw new IllegalArgumentException("The class (" + clazz
+                    + ") has " + memberList.size() + " members (" + memberList + ") with a "
```

Comment:
An object with two or more PlanningIds is an invalid argument regardless of the state (contents of the map).